src/core_plugins/kibana/public/dashboard/grid/dashboard_grid.js

```diff
@@ -207,6 +222,15 @@ export class DashboardGrid extends React.Component {
   }
 
   render() {
+    if (this.state.isLayoutInvalid) {
+      toastNotifications.addDanger({
```

Comment:
This feels like it shouldn't go into render. Render should be non-side-effecting as much as possible, as it can be called numerous times. Not sure where it should go, but somewhere that runs only once, and where side-effects are generally not so frowned upon.

Reply:
++ to this.  I think this might also be causing two error messages to appear when trying to open one dashboard:
<img width="1286" alt="screen shot 2018-03-28 at 9 29 03 am" src="https://user-images.githubusercontent.com/16563603/38032301-4f44f616-326b-11e8-95e3-9ba447c611ef.png">

Maybe just in the constructor?

Reply:
I can move the logic into the constructor but the `render` method still gets called. There is no way to have react stop the render method from getting called.

I could also move the logic for creating the layout into DashboardGridContainer.mapStateToProps function.

thoughts?

Reply:
hmmm, not sure how I feel about it in mapStateToProps.  I've been trying to keep the logic minimal in them, but on the other hand it would be an even earlier spot to catch the error.  I don't feel strongly either way, and if you don't feel strongly either way, maybe just take the easiest course of action and do the redirect inside constructor here.  I think it's okay that this renders.  Maybe just

```
if (this.state.isLayoutInvalid) return null;
```